ext/sqlite3/sqlite3.c

```diff
@@ -2028,6 +2028,10 @@ PHP_METHOD(sqlite3result, fetchArray)
 		RETURN_THROWS();
 	}
 
+	if (sqlite3_column_count(result_obj->stmt_obj->stmt) == 0) {
+		return;
```

Comment:
I think we should `RETURN_FALSE` here, and also set `result_obj->complete = 1`. What do you think?

Reply:
Can you take a look on that commit?
 https://github.com/frikiluser/php-src/commit/a7529d63026b3cca7c9ce3d6acff00bdd4573f54

Changes from prev version: on top of php-7.3 branch, RETURN_FALSE, and completely whipped out result->complete handling.

I'll try to write test cases as requested.

Reply:
Well, the existing `->complete` handling doesn't look quite right (apparently, the `SQLITE_DONE` cases in `::query()` and `::execute()` don't set it), but replacing it by checking `sqlite3_column_count()` for `0` seems to change its semantics; checking for `sqlite3_data_count()` instead might be a good solution, but that function is only available as of sqlite3 3.6.10/3.7.2, but PHP 7.3 is supposed to be compatible with sqlite3 >= 3.3.9. So at least for PHP 7.3 we have to stick with `->complete`, I think. Also we cannot remove the `complete` member for ABI BC reasons in PHP 7.4, but we just could stop using it there. Anyhow, it seems to me that this is a different issue from bug #64531, so likely deserves its own ticket in the bugtracker.

Reply:
I've just pushed a commit with requested changes.
Test included showing what is fixed.

Here: https://github.com/frikiluser/php-src/commit/e8f8af30ad94442cbdd88d715b006c2809aa3eb5

Thanks for reviewing!

Reply:
Thanks for following up on this! It seems, though, that you forgot to push the test case. Could you please have a look?

Reply:
Hi.

I think it's already on my commit: https://github.com/frikiluser/php-src/commit/e8f8af30ad94442cbdd88d715b006c2809aa3eb5

New file: ext/sqlite3/tests/sqlite3result_fetchArray_nonrows_problem.phpt

Now I've updated the PR branch. Sorry for the mess up :-p

Reply:
> […] checking for sqlite3_data_count() instead might be a good solution, but that function is only available as of sqlite3 3.6.10/3.7.2 […]

Correction: sqlite3_data_count() is available at least as of 3.3.9 (likely even longer).